server/src-lib/Hasura/GraphQL/Transport/WebSocket.hs

```diff
@@ -457,13 +458,15 @@ onMessage authMode serverEnv wsConn msgRaw =
                            wsConn authMode params
       CMStart startMsg  -> liftIO $ onStart serverEnv wsConn startMsg
       CMStop stopMsg    -> liftIO $ onStop serverEnv wsConn stopMsg
+      -- TODO do we need onStop/removeLiveQuery cleanup here? I guess this is handled by 'onClose'?
```

Comment:
> I guess this is handled by 'onClose'

That's correct.

Reply:
Under what conditions might `onClose` not get run? I haven't traced this. From what I recall that depends somewhat on internals of the websockets lib we use

Reply:
> From what I recall that depends somewhat on internals of the websockets lib we use

Yes. The `onClose` handler will be run whenever a `WS.ConnectionException` is received, so if there are cases where this exception does not occur but the connection disappears, then `onClose` wouldn't be called.